src/index.js

```diff
@@ -216,22 +217,31 @@ class FsDatastore extends Adapter {
   }
 
   async * _all (q) { // eslint-disable-line require-await
-    // glob expects a POSIX path
-    const prefix = q.prefix || '**'
-    const pattern = path
-      .join(this.path, prefix, '*' + this.opts.extension)
+    let prefix = q.prefix || '**'
+
+    // strip leading slashes
+    prefix = prefix.replace(/^\/+/, '')
+
+    const pattern = `${prefix}/*${this.opts.extension}`
       .split(path.sep)
       .join('/')
-    const files = glob.sync(pattern)
+    const files = glob(this.path, pattern, {
+      absolute: true
+    })
 
     if (!q.keysOnly) {
-      yield * map(files, async (f) => {
-        const buf = await fsReadFile(f)
-        return {
-          key: this._decode(f),
-          value: buf
+      for await (const file of files) {
+        try {
+          const buf = await fsReadFile(file)
+
+          yield {
+            key: this._decode(file),
+            value: buf
+          }
+        } catch (err) {
+          console.info(err)
```

Comment:
🚨 It looks like this would start swallowing errors, which seems unrelated to `Uint8Array` change.
📝 Should this be `console.error(error)` (or `console.warn`) instead ?

Reply:
Oops, no, this shouldn't be in there at all.  It's related to a new test added to interface-datastore that mutates the datastore while running a query (a block is deleted).  To pass the test we don't have to take the mutation into account but we shouldn't explode either.